File: uva10038.py
while True:
    try:
        a=list(map(int,input().split()))
        f=0
        g=[i+1 for i in range(a[0]-1)]
        g2 =[]
        for i in range(1,len(a)-1):
            b=abs(a[i]-a[i+1])
            if  b> a[0]-1 or b<1:
                f=1
                break
            else:
                g2.append(b)
        h=sorted(g2)
        if f==0 and g==h:
            print("Jolly")
        else:
            print("Not jolly")
    except:
        break

File: uva10038.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main()
{
    

    int a;
    // cin>>a;
    while (cin>>a)
    {

        vector<int> count;
        vector<int> num;
        for (int i = 0; i < a ; i++)
        {
            
            int b;
            cin >> b;
            num.push_back(b);
        }

        int f = 0;
        for(int i=0;i<a-1;i++){
            count.push_back(0);
        }
        for (int i = 0; i < a - 1; i++)
        {
            int b = abs(num[i] - num[i + 1]);

            if (b > a-1||b<1)
            {
                f = 1;
                break;
            }
            else
            {
                count[b - 1] += 1;
            }
        }
        for (int i = 0; i < a-1; i++)
        {
            if (count[i] != 1)
            {
                f = 1;
                break;
            }
        }

        if (f == 0)
        {
            cout << "Jolly" << endl;
        }
        else
        {
            cout << "Not jolly" << endl;
        }
    }
}